comp/otelcol/otlp/components/exporter/serializerexporter/factory.go

```diff
@@ -223,12 +217,6 @@ func (f *factory) createMetricExporter(ctx context.Context, params exp.Settings,
 					return err
 				}
 			}
-			if f.wg != nil {
-				f.wg.Wait() // wait for consumeStatsPayload to exit
-			}
-			if f.statsIn != nil {
-				close(f.statsIn)
-			}
```

Comment:
There are moved to datadogexporter/factory since synchronization should happen there